compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -755,6 +755,10 @@ fn ty_is_local_constructor(tcx: TyCtxt<'_>, ty: Ty<'_>, in_crate: InCrate) -> bo
 
         ty::Adt(def, _) => def_id_is_local(def.did(), in_crate),
         ty::Foreign(did) => def_id_is_local(did, in_crate),
+        ty::TyAlias(did, _) => {
+            let ty = tcx.type_of(did);
```

Comment:
I think this needs to be `bound_type_of().subst()`

Reply:
I'm pretty sure you want to use raw `type_of` basically never. when you do `type_of` on something like 

```
type Foo<T> = T;
```

and you're checking, for example, the coherence of `Foo<i32>`, you're going to be instead checking that the def-id of `T` is local, not that `i32` is local.

Please double check the other places you're using `type_of` similarly, I think

Reply:
Oh I see. I'll update it then.